### PR TITLE
fix: coupon errors

### LINKED ISSUE #869: Invalid signature - gift page
**Describe the bug**

I have this issue when I want to register a domain using a new unique code, while I well connected my wallet.

Issue is providing from the lack of money on the paymaster contract.

Capture d’écran 2024-08-08 à 12 55 48

### LINKED ISSUE #870: "Used coupon" - Gift page
**Describe the bug**
Everytime, even if you not used the coupon, we have this information "coupon already used" without doing anything. It's really bad for user experience and people need to ask another coupon (when they can).

### PR DESCRIPTION
**Pull request by Marchand-Nicolas**
BACK PR: https://github.com/starknet-id/api.starknet.id/pull/95
close: #870 
close: #869 

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for transaction issues, providing clearer user feedback with specific error messages.
	- Simplified coupon validation process for improved efficiency.
	- Added loading state feedback during data fetching to inform users.

- **Style**
	- Introduced a new CSS class for styling error messages, enhancing visibility and accessibility.
	- Minor adjustments to existing styles for better layout and text handling.

- **Bug Fixes**
	- Improved error message display during loading states for a more dynamic user experience.

### CODE REVIEW

**Comment by Marchand-Nicolas** [comment]
BACK PR: https://github.com/starknet-id/api.starknet.id/pull/95

File: hooks/paymaster.tsx [hunk 1]
```diff
@@ -51,6 +51,8 @@ const usePaymaster = (
   const { isDeployed, deploymentData } = isStarknetDeployed(account?.address);
   const [deploymentTypedData, setDeploymentTypedData] = useState<TypedData>();
   const [invalidTx, setInvalidTx] = useState<boolean>(false);
+  const [txErrorMessage, setTxErrorMessage] = useState<string>();
+  const [txShortErrorMessage, setTxShortErrorMessage] = useState<string>();
```
Comment (fricoben) [inline hooks/paymaster.tsx:60]: I think you can find a better way to manage the error messages than having to setter for it

**Review by fricoben — CHANGES_REQUESTED** [review]
Working well E2E wise. Just a small clean code demand

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**